specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metricDefinitions_API.json

```diff
@@ -0,0 +1,457 @@
+      "type": "object",
+      "description": "Metric definition class specifies the metadata for a metric.",
+      "properties": {
+        "isDimensionRequired": {
+          "type": "boolean",
+          "description": "Flag to indicate whether the dimension is required."
+        },
+        "resourceId": {
+          "type": "string",
+          "description": "the resource identifier of the resource that emitted the metric."
+        },
+        "namespace": {
+          "type": "string",
+          "description": "the namespace the metric belongs to."
+        },
+        "name": {
+          "$ref": "#/definitions/LocalizableString",
+          "description": "the name and the display name of the metric, i.e. it is a localizable string."
+        },
+        "displayDescription": {
+          "type": "string",
+          "description": "Detailed description of this metric."
+        },
+        "category": {
+          "type": "string",
+          "description": "Custom category name for this metric."
+        },
+        "metricClass": {
+          "$ref": "#/definitions/MetricClass",
+          "description": "The class of the metric."
+        },
+        "unit": {
+          "$ref": "#/definitions/MetricUnit",
+          "description": "the unit of the metric."
+        },
+        "primaryAggregationType": {
+          "$ref": "#/definitions/MetricAggregationType",
+          "description": "the primary aggregation type value defining how to use the values for display."
+        },
+        "supportedAggregationTypes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MetricAggregationType"
+          },
+          "description": "the collection of what aggregation types are supported."
+        },
+        "metricAvailabilities": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MetricAvailability"
+          },
+          "description": "the collection of what aggregation intervals are available to be queried."
+        },
+        "id": {
+          "type": "string",
+          "description": "the resource identifier of the metric definition."
+        },
+        "dimensions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LocalizableString"
+          },
+          "description": "the name and the display name of the dimension, i.e. it is a localizable string."
+        }
+      }
+    },
+    "SubscriptionScopeMetricDefinitionCollection": {
+      "type": "object",
+      "description": "Represents collection of metric definitions.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SubscriptionScopeMetricDefinition"
+          },
+          "description": "The values for the metric definitions."
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "ErrorContract": {
```

Comment:
It seems also that `ErrorContract` is defined in more than one file, please consider moving this object to a common place and referencing it from this new file. The places where it is defined again are:
* specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metrics_API.json
* specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/operations_API.json

Reply:
Created a commonMonitoringTypes.json file and moved it there.